src/shared/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect } from 'react'
+import { Portal } from '@/components'
+import { CSSTransition } from 'react-transition-group'
+import {
+  StyledBackdrop,
+  StyledContainer,
+  StyledTitleText,
+  StyledContentText,
+  StyledHeadRow,
+  StyledButtonContainer,
+  StyledPrimaryButton,
+  StyledSecondaryButton,
+  StyledExitButton,
+  dialogTransitions,
+} from './Dialog.style'
+import { Icon } from '@/shared/components'
+
+type MouseEvent = React.MouseEvent<HTMLButtonElement>
+
+export type DialogProps = {
+  title?: string
+  content: string
+  primaryButton?: string
+  secondaryButton?: string
+  icon?: 'success' | 'failure'
+  exitButton?: boolean
+  showDialog?: boolean
+  handleExit?: (e: MouseEvent) => void
+  handlePrimaryButton?: (e: MouseEvent) => void
+  handleSecondaryButton?: (e: MouseEvent) => void
+}
+
+const Dialog: React.FC<DialogProps> = ({
+  title,
+  content,
+  primaryButton,
+  secondaryButton,
+  icon,
+  exitButton,
+  showDialog,
+  handleExit,
+  handlePrimaryButton,
+  handleSecondaryButton,
+}) => {
+  useEffect(() => {
+    if (!showDialog) {
+      return
+    }
+    document.body.style.setProperty('overflow', 'hidden')
+    return () => {
+      document.body.style.setProperty('overflow', null)
+    }
+  }, [showDialog])
+
+  return (
+    <Portal>
+      <CSSTransition in={showDialog} timeout={250} classNames="modal" unmountOnExit>
+        <StyledBackdrop css={dialogTransitions}>
```

Comment:
Not sure if it's required in this task, but in my opinion, a user should have the ability to close the dialog when he clicks this overlay